package config;

message Config {
  optional float top = 1;
  optional float bottom = 2;
  optional float step_down = 3 [default = 1];
  optional float step_over = 4 [default = 2];
  optional float vertical_tolerance = 5 [default = 0.05];
  optional float tool_diameter = 6 [default = 3.175];
  optional float clearance_above_top = 7 [ default = 4 ];
  optional float engage_above_top = 8 [ default = 2 ];
  optional float feed = 9;
  optional float plunge_feed = 10;
  optional float rotate_x = 11 [default = 0];
  optional float rotate_y = 12 [default = 0];
  optional float rotate_z = 13 [default = 0];
  optional string in_filename = 14;
  optional string out_filename = 15;
  optional bool machine_top = 16 [default = false];
  optional bool waterline_only = 17 [default = false];
  // Forces waterline computation at each step-down level.
  optional bool oblique_approximation = 18 [default = false];
}
